corehq/apps/hqadmin/forms.py

```diff
@@ -117,6 +95,75 @@ def __init__(self, can_toggle_is_staff, *args, **kwargs):
         )
 
 
+class OffboardingUserListForm(forms.Form):
+    csv_email_list = forms.CharField(
+        label="Comma seperated email addresses",
+        widget=forms.Textarea()
+    )
+    invert_apply = forms.BooleanField(
+        label="Invert",
+        initial=False,
+        required=False,
+        help_text=(
+            "Get dimagi.com accounts not included in the above, comma-separated list."
+        )
+    )
+
+    def clean(self):
+        invert = self.cleaned_data.get('invert_apply', False)
+        return clean_data(self.cleaned_data, invert=invert, get_list=True)
+
+    def __init__(self, *args, **kwargs):
+        super(OffboardingUserListForm, self).__init__(*args, **kwargs)
+        self.helper = FormHelper()
+        self.helper.form_class = "form-horizontal"
+        self.helper.label_class = 'col-sm-3 col-md-2'
+        self.helper.field_class = 'col-sm-9 col-md-8 col-lg-6'
+        self.helper.layout = crispy.Layout(
+            'csv_email_list',
+            'invert_apply',
+            FormActions(
+                crispy.Submit(
+                    'superuser_management',
+                    'Get Users'
+                )
+            )
+        )
+
+
+def clean_data(cleaned_data, invert=False, get_list=False):
+    from email.utils import parseaddr
+    from django.contrib.auth.models import User
+    csv_email_list = cleaned_data.get('csv_email_list', '')
+    csv_email_list = re.split(',|\n', csv_email_list)
+    csv_email_list = [parseaddr(em)[1] for em in csv_email_list]
+
+    if not get_list and len(csv_email_list) > 10:
+        raise forms.ValidationError(
+            "This command is intended to grant superuser access to few users at a time. "
+            "If you trying to update permissions for large number of users consider doing it via Django Admin"
+        )
+
+    users = []
+    for username in csv_email_list:
+        if not get_list:
+            if settings.IS_DIMAGI_ENVIRONMENT and "@dimagi.com" not in username:
+                raise forms.ValidationError("Email address '{}' is not a dimagi email address".format(username))
+        try:
+            users.append(User.objects.get(username=username))
+        except User.DoesNotExist:
+            raise forms.ValidationError(
+                "User with email address '{}' does not exist on "
+                "this site, please have the user registered first".format(username))
+
+    if invert:
+        all_users = User.objects.filter(Q(is_superuser=True) | Q(is_staff=True))
+        users = [user for user in all_users if user not in users]
+
+    cleaned_data['users'] = users
```

Comment:
I'm not sure I understand what makes the new field a smell but I've changed it so that the the sanitized list of users overwrites the previously existing `csv_email_list` field. I do agree that each form could have a separated `get_users` method that invoke a shared `clean_csv_email_list` method though.